.pre-commit-config.yaml

```diff
@@ -143,3 +143,8 @@ repos:
         types: [python]
         pass_filenames: false
         args: [".", "-s", "ci", "-s", "python/ray/thirdparty_files", "-s", "python/build", "-s", "lib"]
+  - repo: https://github.com/MarcoGorelli/cython-lint
+    rev: v0.16.2
+    hooks:
+    -   id: cython-lint
+    -   id: double-quote-cython-strings
```

Comment:
we do not run any pre-commit tests on CI. do you want to run this on CI?

how many failures do this lint have right now?

Reply:
CI already has lint for cython, that is flake8 I think. But my laptop flake8 can't handle cython. Just add a newline in `_raylet.pyx`, I tried `ci/lint/format.sh` and got this error:

```sh
~/gits/ray % ci/lint/format.sh
From https://github.com/ray-project/ray
 * branch                  master     -> FETCH_HEAD
Skipped 9 files
All done! ✨ 🍰 ✨
10 files left unchanged.
Traceback (most recent call last):
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 478, in run_ast_checks
    ast = self.processor.build_ast()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/processor.py", line 225, in build_ast
    return ast.parse("".join(self.lines))
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 7
    from cpython.exc cimport PyErr_CheckSignals
                     ^^^^^^^
SyntaxError: invalid syntax

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ruiyangwang/anaconda3/bin/flake8", line 8, in <module>
    sys.exit(main())
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/main/cli.py", line 22, in main
    app.run(argv)
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/main/application.py", line 363, in run
    self._run(argv)
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/main/application.py", line 351, in _run
    self.run_checks()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/main/application.py", line 264, in run_checks
    self.file_checker_manager.run()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 323, in run
    self.run_serial()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 307, in run_serial
    checker.run_checks()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 589, in run_checks
    self.run_ast_checks()
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 480, in run_ast_checks
    row, column = self._extract_syntax_information(e)
  File "/Users/ruiyangwang/anaconda3/lib/python3.10/site-packages/flake8/checker.py", line 465, in _extract_syntax_information
    lines = physical_line.rstrip("\n").split("\n")
AttributeError: 'int' object has no attribute 'rstrip'
```

So I found when I use cython-lint it can lint my cython files, so it can pass the flake8 running in CI

Reply:
I don't have this issue.

this is what I get after adding a blank line at the end of `_raylet.pyx`:

````
$ bash ci/lint/format.sh
$ bash ci/lint/format.sh
INFO: Ray uses shellcheck for shell scripts, which is not installed. You may install shellcheck=0.7.1 with your system package manager.
WARNING: clang-format is not installed!
From https://github.com/ray-project/ray
 * branch                  master     -> FETCH_HEAD
python/ray/_raylet.pyx:5204:1: W391 blank line at end of file
````

on your laptop, you need to run in a conda py3.9 env, and:

````
pip install flake8-quotes flake8-bugbear flake8==3.9.1
````

Reply:
ok, it seems py3.9 works but my workspace defaults to py3.10 where it does not work...

Reply:
> ok, it seems py3.9 works but my workspace defaults to py3.10 where it does not work...

so do you still want to merge this?

if it not exercised on ci, it will just break/drift overtime and become unmaintained.